File: app/src/main/java/com/example/ruslanyussupov/popularmovies/model/Movie.java
package com.example.ruslanyussupov.popularmovies.model;


import android.os.Parcel;
import android.os.Parcelable;

public class Movie implements Parcelable {

    private int id;
    private String originalTitle;
    private String posterPath;
    private String overview;
    private double voteAverage;
    private String releaseDate;
    private String backdropPath;
    private String posterLocalPath;
    private String backdropLocalPath;

    public Movie(int id, String originalTitle, String posterPath, String overview, double voteAverage,
                 String releaseDate, String backdropPath) {
        this.id = id;
        this.originalTitle = originalTitle;
        this.posterPath = posterPath;
        this.overview = overview;
        this.voteAverage = voteAverage;
        this.releaseDate = releaseDate;
        this.backdropPath = backdropPath;
    }

    public Movie(int id, String originalTitle, String posterPath, String overview, double voteAverage,
                 String releaseDate, String backdropPath, String posterDbPath, String backdropDbPath) {
        this.id = id;
        this.originalTitle = originalTitle;
        this.posterPath = posterPath;
        this.overview = overview;
        this.voteAverage = voteAverage;
        this.releaseDate = releaseDate;
        this.backdropPath = backdropPath;
        this.posterLocalPath = posterDbPath;
        this.backdropLocalPath = backdropDbPath;
    }


    protected Movie(Parcel in) {
        id = in.readInt();
        originalTitle = in.readString();
        posterPath = in.readString();
        overview = in.readString();
        voteAverage = in.readDouble();
        releaseDate = in.readString();
        backdropPath = in.readString();
        posterLocalPath = in.readString();
        backdropLocalPath = in.readString();
    }

    public static final Creator<Movie> CREATOR = new Creator<Movie>() {

        @Override
        public Movie createFromParcel(Parcel in) {
            return new Movie(in);
        }

        @Override
        public Movie[] newArray(int size) {
            return new Movie[size];
        }
    };

    public int getId() {
        return id;
    }

    public String getOriginalTitle() {
        return originalTitle;
    }

    public String getPosterPath() {
        return posterPath;
    }

    public String getOverview() {
        return overview;
    }

    public double getVoteAverage() {
        return voteAverage;
    }

    public String getReleaseDate() {
        return releaseDate;
    }

    public String getBackdropPath() {
        return backdropPath;
    }

    public String getPosterLocalPath() {
        return posterLocalPath;
    }

    public String getBackdropLocalPath() {
        return backdropLocalPath;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(id);
        dest.writeString(originalTitle);
        dest.writeString(posterPath);
        dest.writeString(overview);
        dest.writeDouble(voteAverage);
        dest.writeString(releaseDate);
        dest.writeString(backdropPath);
        dest.writeString(posterLocalPath);
        dest.writeString(backdropLocalPath);
    }
}

File: app/src/main/java/com/example/ruslanyussupov/popularmovies/data/model/Movie.kt
package com.example.ruslanyussupov.popularmovies.data.model


import android.os.Parcelable
import androidx.room.Entity
import androidx.room.PrimaryKey

import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

@Parcelize
@Entity(tableName = "movies")
data class Movie(@PrimaryKey val id: Int,
                 @SerializedName("original_title")  val originalTitle: String?,
                 @SerializedName("poster_path")     val posterPath: String?,
                 @SerializedName("overview")        val overview: String?,
                 @SerializedName("vote_average")    val voteAverage: Double?,
                 @SerializedName("release_date")    val releaseDate: String?,
                 @SerializedName("backdrop_path")   val backdropPath: String?,
                 var posterLocalPath: String? = null,
                 var backdropLocalPath: String? = null) : Parcelable {

    val fullPosterPath: String
        get() = MOVIE_POSTER_MAIN_PATH + posterPath

    val fullBackdropPath: String
        get() = MOVIE_BACKDROP_MAIN_PATH + backdropPath

    companion object {
        private const val MOVIE_POSTER_MAIN_PATH = "https://image.tmdb.org/t/p/w185"
        private const val MOVIE_BACKDROP_MAIN_PATH = "https://image.tmdb.org/t/p/w780"
    }

}
